From de32110f32933b2eb0fc2dbe23951a7771b0941f Mon Sep 17 00:00:00 2001
From: Sergey Larin <cerg2010cerg2010@mail.ru>
Date: Sat, 29 Jun 2019 15:30:57 +0300
Subject: [PATCH] ARM: dts: tegra20-glide: Set up VSENSOR regulator

This is not configured in downstream kernel, but it was identified that
disabling this regulator makes these sensors not working (note: kxtf9
included).

Signed-off-by: Sergey Larin <cerg2010cerg2010@mail.ru>
---
 arch/arm/boot/dts/tegra20-glide.dts | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/tegra20-glide.dts b/arch/arm/boot/dts/tegra20-glide.dts
index 74edc637f4d1..c3698603ee69 100644
--- a/arch/arm/boot/dts/tegra20-glide.dts
+++ b/arch/arm/boot/dts/tegra20-glide.dts
@@ -512,6 +512,8 @@
 			mount-matrix = "0", "1", "0",
 				"-1", "0", "0",
 				"0", "0", "-1";
+			vdd-supply = <&vsensor>;
+			vlogic-supply = <&vsensor>;
 
 			i2c-gate {
 				#address-cells = <1>;
@@ -729,12 +731,10 @@
 					regulator-max-microvolt = <3300000>;
 				};
 
-				ldo10 {
+				vsensor: ldo10 {
 					regulator-name = "nvvdd_ldo10,vsensor_1v8";
 					regulator-min-microvolt = <1800000>;
 					regulator-max-microvolt = <1800000>;
-					regulator-always-on;
-					regulator-boot-on;
 				};
 
 				ldo11 {
@@ -893,6 +893,7 @@
 			compatible = "asahi-kasei,ak8975";
 			reg = <0xc>;
 			gpios = <&gpio TEGRA_GPIO(K, 4) GPIO_ACTIVE_HIGH>;
+			vdd-supply = <&vsensor>;
 		};
 	};
 
-- 
2.22.0

